### PR TITLE
#168131525 Users should be able to Login to Barefoot Nomad

### PR DESCRIPTION
**Pull request by Dkabiswa**
#### What does this PR do?
- Adds local login feature
- Adds social Auth login
#### Description of Task to be completed?
- Get user input
- Login the user if the password and email are valid
- Give user social Auth links
- Redirect the user to the logged-in page if valid authentication 
#### How should this be manually tested?
- Git clone this repo and checkout this branch
- Run `npm install`
- Run` npm start` to start client-server
- login in with these correct credentials 
```
emali : johndoe@gmail.com
password: Root1123#
```
- Or continue with either google or facebook
- Run `npm run storybook:dev` to view components in the story book
#### Any background context you want to provide?
- To continue with facebook you need a test fb account
- logout of your account or open incognito browser
- you can use
```
email: davis_bppcoen_kabiswa@tfbnw.net
password: mervels
```
#### What are the relevant pivotal tracker stories?
[168131525](https://www.pivotaltracker.com/story/show/168131525)
[168131526](https://www.pivotaltracker.com/story/show/168131526)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: src/components/shared/input.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+// Input.defaultProps = {
+//   name: '',
+//   type: 'text',
+//   className: 'input',
+//   placeholder: '',
+//   value: '',
+// };
+
+// export default Input;
```
Comment (willywax) [inline src/components/shared/input.js]: Remove the commented code

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "eslint.autoFixOnSave": true
+}
```
Comment (drayzii) [inline .vscode/settings.json]: Can you add this file to `.gitignore`

Comment (drayzii) [reply .vscode/settings.json]: And both `.DS_Store`s

File: src/components/LoginPage.js [hunk 3]
```diff
@@ -1,43 +1,124 @@
...
+                inputType="password"
+                placeholder="Password"
+                onChange={this.handleInput}
+                value={password}
+                required
+              />
+            <div className="forgot">
+                <a  href="/">Forgot your password?</a>
+            </div>
+            <button className="btnn btnn-primary log" type="submit">Login</button>
```
Comment (drayzii) [inline src/components/LoginPage.js]: What is the reason behind using `btnn` for buttons

File: src/components/shared/socialAuth.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+/* eslint-disable react/forbid-prop-types */
+import React from 'react';
+import google from '../../logo/iconfinder_Google_1298745.png';
+import fb from '../../logo/iconfinder_square-facebook_317727.png';
+
+const socialAuth = () => {
+  const baseUrl = 'https://mervels-bn-backend-staging.herokuapp.com';
```
Comment (drayzii) [inline src/components/shared/socialAuth.js]: I think this can be imported from `/src/config`

File: .vscode/settings.json (hunk 2 above)
Comment (Dkabiswa) [reply .vscode/settings.json]: okay

File: src/components/LoginPage.js (hunk 3 above)
Comment (Dkabiswa) [reply src/components/LoginPage.js]: Ineeded to add some more styles for the anchor tags was changing to buttons and didn't want to collide with the general buttons but it's specific to only those

File: src/components/shared/socialAuth.js (hunk 4 above)
Comment (Dkabiswa) [reply src/components/shared/socialAuth.js]: okay

**Pull request merged**